### PR TITLE
Unify spire setup

### PR DESCRIPTION
**Pull request by d-uzlov**
## Description

This PR moves spire setup for multi-cluster and k8s_monolith examples into `examples/spire`.

Single-cluster and monolith examples share `spire/single-cluster` setup.
Multi-cluster examples share `spire/clusterX` setups.

In `examples/interdomain`, `examples/floating_interdomain`, `examples/k8s_monolith/external_nsc`, `examples/k8s_monolith/external_nse` there are new `spiffe-federation` setups, that contain scenario-specific setup.

## Issue link

https://github.com/networkservicemesh/deployments-k8s/issues/7673

## How Has This Been Tested?

- [ ] Added unit testing to cover
- [x] Tested manually
- [x] Tested by integration testing
- [ ] Have not tested

## Types of changes

- [ ] Bug fix
- [ ] New functionallity
- [ ] Documentation
- [x] Refactoring
- [ ] CI

### CODE REVIEW

File: examples/spire/base/kustomization.yaml [hunk 1]
```diff
@@ -4,22 +4,22 @@ kind: Kustomization
 
 namespace: spire
 
-generatorOptions:
-  disableNameSuffixHash: true
+# generatorOptions:
```
Comment (glazychev-art) [inline examples/spire/base/kustomization.yaml:7]: Do we need to remove this?

Comment (d-uzlov) [reply examples/spire/base/kustomization.yaml]: Fixed

**Review by denis-tingaikin — APPROVED** [review]
LGTM

@d-uzlov Could you rebase this?

**Comment by d-uzlov** [comment]
@denis-tingaikin, done.

**Review by glazychev-art — APPROVED** [review]

**Pull request merged**